frame/bribedao/src/lib.rs

```diff
@@ -0,0 +1,315 @@
+		pub fn create_bribe(
+			origin: OriginFor<T>,
+			request: CreateBribeRequest<T>,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+			// Freeze/hold the users funds, to verify payment
+			let holdreq = request.clone();
+			T::Currency::hold(holdreq.asset_id, &from, holdreq.total_reward)
+				.map_err(|_| Error::<T>::CantFreezeFunds)?; //Freeze assets
+
+			let id = <Self as Bribe>::create_bribe(request.clone())?;
+			Self::deposit_event(Event::BribeCreated { id, request });
+			Ok(().into())
+		}
+
+		/// Sell Votes request
+		#[pallet::weight(10_000)]
+		pub fn take_bribe(
+			origin: OriginFor<T>,
+			request: TakeBribeRequest<T>,
+		) -> DispatchResultWithPostInfo {
+			let _from = ensure_signed(origin)?;
+			let bribe_index = request.bribe_index;
+			let bribe_taken = <Self as Bribe>::take_bribe(request.clone())?;
+			if bribe_taken {
+				Self::deposit_event(Event::BribeTaken { id: bribe_index, request });
+			}
+			Ok(().into())
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		InvalidBribe,
+		InvalidIndex,
+		NotEnoughFunds,
+		NotEnoughStake,
+		PriceNotRequested,
+		AlreadyBribed,
+		EmptySupply,
+		CantFreezeFunds,
+		ReleaseFailed,
+		BribeDeletionFailed,
+		InvalidVote,
+	}
+
+	// offchain indexing
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn offchain_worker(_b: T::BlockNumber) {
+			log::info!("Indexing request offchain");
+		}
+	}
+
+	impl<T: Config> Bribe for Pallet<T> {
+		type BribeIndex = BribeIndex;
+		type AccountId = T::AccountId;
+		type ReferendumIndex = ReferendumIndex;
+		type Balance = T::Balance;
+		type Conviction = T::Conviction;
+		type CurrencyId = T::CurrencyId;
+
+		/// Register new bribe request
+		fn create_bribe(request: CreateBribeRequest<T>) -> Result<Self::BribeIndex, DispatchError> {
+			Self::do_create_bribe(request)
+		}
+
+		/// Register the votes a user wants to sell
+		fn take_bribe(request: TakeBribeRequest<T>) -> Result<bool, DispatchError> {
+			Self::do_take_bribe(request)
+		}
+
+		/// Delete a finished Bribe Request
+		fn delete_bribe(request: DeleteBribeRequest) -> Result<bool, DispatchError> {
+			// todo, make sure this function can not be abused
+			Self::do_delete_bribe(request)
+			//			Ok(true)
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Create a new Bribe Request
+		fn do_create_bribe(request: CreateBribeRequest<T>) -> Result<BribeIndex, DispatchError> {
+			let id = BribeCount::<T>::mutate(|id| {
+				*id += 1;
+				*id
+			});
+
+			ensure!(!BribeRequests::<T>::contains_key(id), Error::<T>::AlreadyBribed); //dont duplicate briberequest if we already have it
+
+			BribeRequests::<T>::insert(id, request);
+			Ok(id)
+		}
+
+		/// Find votes for a bribe request
+		fn do_match_votes(bribe_index: BribeIndex) -> Result<bool, DispatchError> {
+			ensure!(BribeRequests::<T>::contains_key(bribe_index), Error::<T>::InvalidIndex);
+
+			let bribe_request = BribeRequests::<T>::get(bribe_index).unwrap();
```

Comment:
Unless I am missing something, I think we can merge the two cases using `try_get ... map_err ..`

This will also remove the usage of _unwrap_ here. Which is a good thing :)

Reply:
updated in latest to a try_get from storage map with a map error